src/gl_context.rs

```diff
@@ -0,0 +1,145 @@
+/// will match the GdkVisual of the window.
+///
+/// A GLContext is not tied to any particular normal framebuffer. For instance, it cannot draw to
+/// the GdkWindow back buffer. The GDK repaint system is in full control of the painting to that.
+/// Instead, you can create render buffers or textures and use gdk_cairo_draw_from_gl() in the draw
+/// function of your widget to draw them. Then GDK will handle the integration of your rendering
+/// with that of other widgets.
+///
+/// Support for GLContext is platform-specific, context creation can fail, returning None context.
+///
+/// A GLContext has to be made "current" in order to start using it, otherwise any OpenGL call will
+/// be ignored.
+///
+/// ##Creating a new OpenGL context
+///
+/// In order to create a new GLContext instance you need a GdkWindow, which you typically get during
+/// the realize call of a widget.
+///
+/// A GLContext is not realized until either gdk_gl_context_make_current(), or until it is realized
+/// using gdk_gl_context_realize(). It is possible to specify details of the GL context like the
+/// OpenGL version to be used, or whether the GL context should have extra state validation enabled
+/// after calling gdk_window_create_gl_context() by calling gdk_gl_context_realize(). If the
+/// realization fails you have the option to change the settings of the GLContext and try again.
+///
+/// ##Using a GLContext
+///
+/// You will need to make the GLContext the current context before issuing OpenGL calls; the system
+/// sends OpenGL commands to whichever context is current. It is possible to have multiple contexts,
+/// so you always need to ensure that the one which you want to draw with is the current one before
+/// issuing commands:
+///
+/// ```Rust,ignore
+/// GLContext::make_current(context);
+/// ```
+///
+/// You can now perform your drawing using OpenGL commands.
+///
+/// You can check which GLContext is the current one by using gdk_gl_context_get_current(); you can
+/// also unset any GLContext that is currently set by calling gdk_gl_context_clear_current().
+pub type GLContext = Object<ffi::GdkGLContext>;
+
+impl GLContext {
+    /// Retrieves the Display the context is created for
+    pub fn get_display(&self) -> Option<Display> {
+        unsafe { from_glib_none(ffi::gdk_gl_context_get_display(self.to_glib_none().0)) }
+    }
+
+    /// Retrieves the GdkWindow used by self.
+    pub fn get_window(&self) -> Option<Window> {
+        unsafe { from_glib_none(ffi::gdk_gl_context_get_window(self.to_glib_none().0)) }
+    }
+
+    /// Retrieves the GLContext that this context share data with.
+    pub fn get_shared_context(&self) -> Option<Window> {
+        unsafe { from_glib_none(ffi::gdk_gl_context_get_shared_context(self.to_glib_none().0)) }
+    }
+
+    /// Retrieves the OpenGL version of the context.
+    ///
+    /// The context must be realized prior to calling this function.
+    pub fn get_version(&self, major: &mut i32, minor: &mut i32) {
```

Comment:
The generator will return these in a tuple so to avoid breakage I'd suggest doing the same.
Example: https://github.com/gkoz/gir/pull/122#issuecomment-143478279